### DISCUSSION TITLE
Trigger validation with nested fields

### DISCUSSION THREAD

**Discussion by andrepxa** [react-hook-form/react-hook-form#5337] [Q&A]
Hi,

I have some nested fields in a paginated form and using `trigger` function to validate them when changing tabs.
While trying to update from version 5 to 7, I noticed that `trigger` function is returning `true` even with errors on nested fields.

I wrote a simple form to simulate it with two fields, `firstname` and `last.name`. With just the not nested field filled, the `trigger()` returns `true` but `trigger('last.name')` returns `false`, and the error object is like: 
```js
{
  last: {
    name: {
      type: "required",
      message: "", 
      ref: <input name="last.name" placeholder="Last name"></input>
    }
  }
}
```

[codesandbox example](https://codesandbox.io/s/react-hook-form-js-forked-hpp9l?file=/src/App.js)

Is `trigger()` supposed to validated all fields, even those registered like `{...register('last.name', { required: true })}`, or I got it wrong?
 
Thank you very much for your time.

**Comment by bluebill1049** [comment]
can you provide a codesandbox to reproduce this problem? trigger() should validate the whole form.

**Reply by andrepxa** [reply]
I provided it. Is it not working or is private? 
https://codesandbox.io/s/react-hook-form-js-forked-hpp9l?file=/src/App.js

**Reply by bluebill1049** [reply]
<img width="645" alt="Screen Shot 2021-05-26 at 2 54 26 pm" src="https://user-images.githubusercontent.com/10513364/119604372-462e1700-be32-11eb-887e-f49c8ec05f64.png">
<img width="669" alt="Screen Shot 2021-05-26 at 2 54 37 pm" src="https://user-images.githubusercontent.com/10513364/119604378-4b8b6180-be32-11eb-9f9a-bd9e701fd815.png">

what's the issue above? could you share the steps?

**Reply by andrepxa** [reply]
Enter only the first input and try to trigger
![image](https://user-images.githubusercontent.com/3040200/119668346-7d3e0180-be0d-11eb-911c-7a506c458255.png)

**Reply by bluebill1049** [reply]
I am looking into it.

**Reply by bluebill1049** [reply]
issue: https://github.com/react-hook-form/react-hook-form/issues/5382

**Reply by andrepxa** [reply]
thanks for the support @bluebill1049
